src/configuration_store.cpp

```diff
@@ -2,18 +2,14 @@
 
 namespace eppoclient {
 
-ConfigurationStore::ConfigurationStore() : configuration_(std::nullopt) {}
+ConfigurationStore::ConfigurationStore() : configuration_(nullptr) {}
```

Comment:
minor: given this used to return an empty `Configuration` instead of `null`, shall we use that as default? This would also minimize in other places
```suggestion
ConfigurationStore::ConfigurationStore() : ConfigurationStore(Configuration()) {}
```